src/v/cloud_storage/cache_service.cc

```diff
@@ -9,41 +9,96 @@
  */
 
 #include "cloud_storage/logger.h"
-#include "random/generators.h"
 #include "utils/gate_guard.h"
 #include "vlog.h"
 
 #include <seastar/core/coroutine.hh>
 #include <seastar/core/file.hh>
 #include <seastar/core/fstream.hh>
 #include <seastar/core/future.hh>
-#include <seastar/core/print.hh>
 #include <seastar/core/seastar.hh>
 #include <seastar/core/smp.hh>
-#include <seastar/util/defer.hh>
 
 #include <cloud_storage/cache_service.h>
 
 #include <exception>
 #include <filesystem>
 #include <optional>
-#include <string>
-#include <system_error>
 
 namespace cloud_storage {
 
-cache::cache(std::filesystem::path cache_dir) noexcept
+cache::cache(
+  std::filesystem::path cache_dir,
+  size_t max_cache_size,
+  ss::lowres_clock::duration check_period) noexcept
   : _cache_dir(std::move(cache_dir))
-  , _cnt(0) {}
+  , _max_cache_size(max_cache_size)
+  , _check_period(check_period)
+  , _cnt(0)
+  , _total_cleaned(0) {}
+
+uint64_t cache::get_total_cleaned() { return _total_cleaned; }
+
+ss::future<> cache::clean_up_cache() {
+    gate_guard guard{_gate};
+    co_await _walker.walk(_cache_dir.native());
+
+    if (_walker.current_cache_size >= _max_cache_size) {
+        auto size_to_delete = _walker.current_cache_size
+                              - (_max_cache_size * _cache_size_low_watermark);
+
+        auto candidates_for_deletion = _walker.files;
+
+        uint64_t deleted_size = 0;
+        size_t i_to_delete = 0;
+        while (deleted_size < size_to_delete) {
+            // todo: don't delete .part file that are being written
+            // open file exclusively before deleting
```

Comment:
this comment should probably be removed since you decided not to do this

Reply:
No, I decided to do it :)